synapse/http/matrixfederationclient.py

```diff
@@ -264,14 +264,33 @@ def _request(self, destination, method, path,
             defer.returnValue(response)
 
     def sign_request(self, destination, method, url_bytes, headers_dict,
-                     content=None):
+                     content=None, destination_is=None):
+        """
+        Signs a request by adding an Authorization header to headers_dict
+        Args:
+            destination (str): The desination home server of the request. May be null if the
+                destination is an identity server, in which case destination_is must be non-null.
+            method (str): The HTTP method of the request
+            url_bytes (str): ?
+            headers_dict (dict): Dictionary of request headers to append to
+            content (str): The body of the request
+            destination_is (str): As 'destination', but if the destination is an identity server
+
+        Returns:
+            Deferred
+        """
         request = {
             "method": method,
             "uri": url_bytes,
             "origin": self.server_name,
-            "destination": destination,
         }
 
+        if destination is not None:
+            request["destination"] = destination
+
+        if destination_is is not None:
+            request["destination_is"] = destination_is
```

Comment:
so what is the logic behind putting this in a different field?

Reply:
To make sure it's always distinct from a request to a homeserver with the same name, eg. that you could reuse the same signature for a request to the matrix.org HS to sign the same request to the matrix.org IS as they're different destinations (although in practice are unlikely to support the same request).